internal/resources/resource_host.go

```diff
@@ -509,9 +510,11 @@ func getVAsForHost(hostID string, vas []rest.VolumeAttachment) []rest.VolumeInfo
 func resourceMetalHostUpdate(d *schema.ResourceData, meta interface{}) (err error) {
 	defer func() {
 		var nErr = rest.GenericOpenAPIError{}
+
 		if errors.As(err, &nErr) {
 			err = fmt.Errorf("failed to update host %s: %w", strings.Trim(nErr.Message(), "\n "), err)
-
+		} else if err != nil {
+			err = fmt.Errorf("failed to update host %w", err)
```

Comment:
Any particular reason for this?  I think this is saying we may get some error that's not GenericOpenAPIError.  If so, do we need the same in other resources?

Reply:
I added this hoping that the linter would stop flagging the 'return err' legs with the 'wrapcheck' error. But, that didn't stop the error. Left it in since we now have all non-GenericOpenAPIError errors returned wrapped from the 'defer' function. If this makes sense I can add it to the other resources as well - would do this under DE13042.

Reply:
Yes, I would like to see these defers consistent.  If it makes sense for host, let's make the other resources the same.  Thanks.

Reply:
Mike - will cover the defer functions for all resources with [DE13042](https://rally1.rallydev.com/#/431610598592ud/dashboard?detail=%2Fdefect%2F639010228255). I think this can be written as a separate function and invoked for all resources.